Exadel.Compreface/Services/ExampleSubjectService.cs

```diff
@@ -1,21 +1,54 @@
-﻿using Exadel.Compreface.Configuration;
-using Exadel.Compreface.DTOs.ExampleSubject.AddExampleSubject;
-using Exadel.Compreface.DTOs.ExampleSubject.ListAllExampleSubject;
+﻿using System.Net.Http.Json;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.AddBase64ExampleSubject;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.AddExampleSubject;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteAllSubjectExamples;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteImageById;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteMultipleExamples;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DownloadImageById;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DownloadImageBySubjectId;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.ListAllExampleSubject;
+using Exadel.Compreface.Configuration;
 using Flurl;
 using Flurl.Http;
+using Exadel.Compreface.DTOs.HelperDTOs;
+using Exadel.Compreface.Exceptions;
 
 namespace Exadel.Compreface.Services;
 
 public class ExampleSubjectService
 {
-    private readonly ComprefaceConfiguration _configuration;
-    
+    private readonly IComprefaceConfiguration _configuration;
+
     public ExampleSubjectService(ComprefaceConfiguration configuration)
     {
         _configuration = configuration;
     }
-    
     public async Task<AddExampleSubjectResponse> AddExampleSubject(AddExampleSubjectRequest request)
+    {
+        try
+        {
```

Comment:
I would add Extra layer for Flurl.Http methods, to do that we need to create a static class like `ApiClient` or something. Then we will add `PostJsonAsync` method and other methods as well but in this case with `try/catch` so then we will not need to write `try/catch`  for every request, we will just use `ApiClient` class's extension methods to do the job

Reply:
Are you offer to write extensions for every method like a PostJsonAsync, GetJsonAsync and etc.?